gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/metrics/KafkaReportingFormats.java

```diff
@@ -68,6 +70,10 @@
     switch (this) {
       case AVRO:
         KafkaAvroEventReporter.Builder<?> builder = KafkaAvroEventReporter.Factory.forContext(context);
+        if (properties.getProperty(PusherUtils.KAFKA_PUSHER_CLASS_NAME_KEY,
```

Comment:
Maybe you can remove the dependency on the pusher name by instantiating through a factory name. Something like. ```Class myClass = Class.forName(<factory name>);
        Method method;
        try {
          method = myClass.getDeclaredMethod("forContext", MetricContext.class);
          builder = method.invoke(context);
        } catch (NoSuchMethodException e) {
          try {
            method = myClass.getDeclaredMethod("forContext", MetricContext.class, Properties.class);
            builder = method.invoke(context, properties);
          } catch (NoSuchMethodException e) {
            throw RuntimeException();
          }
        }```